File: svc-event/src/main/java/com/beancounter/event/service/EventBehaviourFactory.java
package com.beancounter.event.service;

import com.beancounter.common.event.CorporateEvent;
import com.beancounter.event.service.alpha.AlphaEventAdapter;
import java.util.HashMap;
import java.util.Map;
import org.springframework.stereotype.Service;

@Service
public class EventBehaviourFactory {
  Map<String, Event> adapters = new HashMap<>();

  public EventBehaviourFactory() {
    adapters.put("ALPHA", new AlphaEventAdapter(new TaxService()));
  }

  public Event getAdapter(CorporateEvent event) {
    return adapters.get(event.getSource());
  }
}

File: svc-event/src/main/kotlin/com/beancounter/event/service/EventBehaviourFactory.kt
package com.beancounter.event.service

import com.beancounter.common.event.CorporateEvent
import com.beancounter.event.service.alpha.AlphaEventAdapter
import org.springframework.stereotype.Service

/**
 * Factory for obtain event behaviour providers.
 */
@Service
class EventBehaviourFactory {
    // final var adapters: MutableMap<String, Event> = HashMap()
    private final val alphaEventAdapter = AlphaEventAdapter(TaxService())

    fun getAdapter(event: CorporateEvent): Event {
        return alphaEventAdapter // We only have one adapter
    }
}